cpukit/include/rtems/score/thread.h

```diff
@@ -835,6 +836,14 @@ struct _Thread_Control {
 
   /** This field contains the context of this thread. */
   Context_Control                       Registers;
+  
+#if defined (USE_THREAD_STACK_PROTECTION)
+  /** This is the control block of the protected stack. */
+  Stackprotection_Stack            the_stack;
+  
+  /** This is the control block for handling stack address names. */
+  Stackprotection_Stack_name      name_block;
```

Comment:
I don't think these structures are terribly useful. You can just define all the relevant pieces here. Note that some stack info is already kept in `Stack_Control` structure and pointed to by the `Thread_Start_information` structure. Why aren't you integrating with those existing pieces?